src/ucp/core/ucp_worker.c

```diff
@@ -573,13 +573,26 @@ ucs_status_t ucp_worker_set_ep_failed(ucp_worker_h worker, ucp_ep_h ucp_ep,
     return ret_status;
 }
 
+static void ucp_worker_discard_wireup_aux_ep(ucp_wireup_ep_t *wireup_ep,
```

Comment:
IMO, since this function is just needed to pass purge_cb/purge_arg as do_assert/NULL to ucp_wireup_ep_discard_aux_ep, better remove it, and call ucp_wireup_ep_discard_aux_ep() directly with (..., do_assert, NULL)

Reply:
yes, i though about this, but there is a long comment
is it ok duplicating the comment then?

Reply:
i think we can shorten or drop the comment
/* Pending queue should be empty */

Reply:
btw, mabe pass ucp_destroyed_ep_pending_purge?

Reply:
nice catch, will do it